versions/1/data_stream/_dev/test/pipeline/expected.spec.yml

```diff
@@ -5,11 +5,16 @@ spec:
   # Everything under here follows JSON schema (https://json-schema.org/), written as YAML for readability
   type: object
   additionalProperties: false
+  definitions:
+    optionalTestResult:
+      anyOf:
+        - type: object
+          additionalProperties: true
+        - type: "null"
   properties:
     expected:
       description: Expected test result
       type: array
       items:
-        type: object
-        additionalProperties: true
+        $ref: "#/definitions/optionalTestResult"
```

Comment:
Nit: why not just inline the definition right here? Is it not possible or did you define it separately for some other reason?

Reply:
Yes, it wasn't possible. It failed with a strange error (trying to figure out if it's array of strings, array of objects, but definitely not null)